gsql/cache/cache_handler.py

```diff
@@ -0,0 +1,107 @@
+import json
+import os
+from json.decoder import JSONDecodeError
+
+
+CacheDict = {}
+
+
+def keys_exists(element, *keys):
+    """
+    Check if *keys (nested) exists in `element` (dict).
+    """
+    if not isinstance(element, dict):
+        raise AttributeError("keys_exists() expects dict as first argument.")
+    if len(keys) == 0:
+        raise AttributeError(
+            "keys_exists() expects at least three arguments, one given."
+        )
+
+    _element = element
+    for key in keys:
+        try:
+            _element = _element[key]
+        except KeyError:
+            return False
+    return True
+
+
+def count_lines_in_json(filename="log_data.json"):
+    with open(filename, "r") as file:
+        try:
+            file_data = json.load(file)
+            return len(file_data)
+        except JSONDecodeError as error:
+            print(error)
+
+
+def check_json_file_exists(filename="log_data.json") -> bool:
+    if os.path.exists(filename):
+        return True
+    else:
+        return False
+
+
+def create_json_file(filename="log_data.json"):
+    with open(filename, "w") as file:
+        print("File created successfully")
+        json.dump({}, file)
+        file.close()
+
+
+def remove_key_from_json(keypair, filename="log_data.json"):
+    if check_json_file_exists() is True:
+        with open(filename, "r") as file:
+            try:
+                file_data = json.load(file)
+                if keys_exists(file_data, keypair) is True:
+                    del file_data[keypair]
+                    print("key removed successfully from file")
+            except JSONDecodeError as error:
+                print(error)
```

Comment:
Do not handle exception here, just raise the exception, the parent class will be responsible for handling all exceptions.

Reply:
Done